rules/rom_keys.bzl

```diff
@@ -0,0 +1,73 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("//rules:const.bzl", "CONST")
+
+# ROM/ROM_EXT key roles
+KEY_ROLE = struct(
+    DEV = "dev",
+    TEST = "test",
+    PROD = "prod",
+    UNAUTHORIZED = "unauthorized",
+)
+
+RomKeyMap = provider(
```

Comment:
Why a `provider`?  Providers are usually used typed inputs and outputs of rules.  The way this is currently being used, `struct` would be more appropriate.

Using a rule to define a key domain is not a bad idea, but we'd have to investigate what sorts of changes would be required in the opentitan_{binary, test, etc} macros are needed to consume a provider.  Using a provider, however, would allow defining a key domain to be a rule in the codebase, like:

//sw/device/silicon_creator/rom/keys/fake/BUILD
```
key_domain(
  name = "fake_rom_keys",
  slots = {
    "test_0": "test_key_0.pem",
    ...
  },
)
```

As things are now in `//rules/opentitan.bzl`, the macro wants to iterate over the set of keys and create the `sign_bin` rule for each one, but macros are not allowed to consume providers (only other rules are).  Without some restructuring of opentitan_{test, flash_binary, etc}, I don't think a provider is appropriate just yet.